### ISSUE TITLE
Understanding accuracy

### ISSUE THREAD

**Issue by Sum02dean** [OpenNMT/OpenNMT-py#1776]
My final train and validation accuracy are 98% and 91% respectively. Yet top-1 accuracy is 50%.
What am I missing here? What is causing this discrepancy between 91% accuracy and 50% top-1 accuracy?

Many thanks.

Dean

**Comment by francoishernandez** [comment]
Hey @Sum02dean 
What is it that you call "top-1 accuracy" here?
It's probably a simple matter of over-fitting to your training data/domain though.

**Comment by Sum02dean** [comment]
@francoishernandez  Top-n accuracy is the percentage that the correct (ground-truth) reaction occurs in the n_best predictions from beam search. I am using openNMT to predict on chemical structures (reactants>products), each atom here is a token. 

I think that the accuracy/val accuracy is the token-token accuracy. While this is good for natural language, a simple miss classification in such "text space" may result in an invalid chemical sequence in "chemical space". I assume this is the  problem.

**Comment by francoishernandez** [comment]
Yes, your result is quite expected. 100% accuracy, like needed in your case, is fairly rare in such sequence to sequence setups.
Training and validation accuracies are just based on the next word prediction from the ground truth, not a full decoding.
